src/index.ts

```diff
@@ -1,24 +1,34 @@
 import { TypedError } from 'typed-error';
+import * as EventEmitter from 'eventemitter3';
 
 export class PromiseQueueError extends TypedError {}
 export class MaxSizeExceededError extends PromiseQueueError {}
 export class TimeoutError extends PromiseQueueError {}
 
+const durationSince = (t0: ReturnType<typeof process.hrtime>): number => {
+	const diff = process.hrtime(t0);
+	return diff[0] * 1000 + diff[1] / 1e6;
+};
+
 export class PromiseQueue {
+	public name: string;
```

Comment:
Why is the name added? It seems like something that would be handled/used purely in dependents code - it's not actually used anywhere in this module

Reply:
True, I can see in the branch on the API that there are "pool" objects where this ought to live.

Reply:
Removed all references to name